src/pragha-utils.c

```diff
@@ -233,30 +233,30 @@ gchar* convert_length_str(gint length)
 	str = g_new0(char, 128);
 	memset(tmp, '\0', 24);
 
-	if (length > 86400) {
+	if (length >= 86400) {
 		days = length/86400;
 		length = length%86400;
 		g_sprintf(tmp, "%d %s, ", days, ngettext("day", "days", days));
 		g_strlcat(str, tmp, 24);
 	}
 
-	if (length > 3600) {
+	if (length >= 3600) {
 		hours = length/3600;
 		length = length%3600;
 		memset(tmp, '\0', 24);
 		g_sprintf(tmp, "%d:", hours);
 		g_strlcat(str, tmp, 24);
 	}
 
-	if (length > 60) {
+	if (length >= 60) {
 		minutes = length/60;
 		length = length%60;
 		memset(tmp, '\0', 24);
 		g_sprintf(tmp, "%02d:", minutes);
 		g_strlcat(str, tmp, 24);
 	}
 	else
-		g_strlcat(str, "00:", 4);
+		g_strlcat(str, "00:", 24);
```

Comment:
> One of the first concatenations that are made in this function, are the days, that their size depends on the translation that we can not control in any way.

Well actually, that's something for translators to take into account. You wouldn't want either to have a string that you cannot display in the interface because it is too long.
So the question is what does "reasonable length" entails, and while 24 might be a bit too short, 128 seems a bit long.

> what benefit does this instead doing a simple g_printf to formating the time?

I am not sure to understand your question correctly, but if you propose to use a single `g_printf` instead of several ones and concatenating the result as we do now, we would still need different format strings depending on which components of the time are zeroed (e.g. you wouldn't display "0 days, 00:01:15" if you could just display "1:15").

However without the buffer we could get rid of the arbitrary length limitation, and the code would probably be a bit cleaner as well. I'll get a try at it.